### PR TITLE
Pa02

### PR DESCRIPTION
**Pull request by Andrehoi**

### CODE REVIEW

**Review by Sebbek8 — APPROVED** [review]

File: src/pa02/snakes_simulations.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        self.game_list = []
+
+        # Applies the board class to each player
+        self.players = []
+        for classes in player_field:
+            player = classes(self.board)
+            self.players.append(player)
+
+        # Randomizes the player order if True
+        if randomize_players:
```
Comment (ahmedalbuni) [inline src/pa02/snakes_simulations.py:171]: Randomize player should be done for each game, not inside the init method

File: src/pa02/snakes_simulations.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+        """
+        return self.counter
+
+
+class Simulation:
+    """
+    Class that simulates the snakes and ladders game with the player and
+    board classes
+    """
+    def __init__(self, player_field=[Player, LazyPlayer],
```
Comment (ahmedalbuni) [inline src/pa02/snakes_simulations.py:158]: Don't use mutables as default arguments

File: src/pa02/snakes_simulations.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+class ResilientPlayer(Player):
+    """
+    Sub-class of player that represents a resilient player
+    """
+    def __init__(self, board, extra_steps=1):
+        super().__init__(board)
+        self.add_move = extra_steps
+        self.chuted = False
+        self.counter = 0
```
Comment (ahmedalbuni) [inline src/pa02/snakes_simulations.py:83]: You could have use the same variable from the parent class

**Review by ahmedalbuni — APPROVED** [review]
Good job!

**Pull request merged**